src/docfx/lib/github/GitHubAccessor.cs

```diff
@@ -162,127 +107,97 @@ public async Task<(Error, IEnumerable<GitHubUser>)> GetUsersByCommit(string owne
+                    .HandleTransientHttpError()
+                    .Or<OperationCanceledException>()
+                    .Or<IOException>()
+                    .RetryAsync(3)
+                    .ExecuteAsync(() => _httpClient.PostAsync(s_url, new StringContent(request, Encoding.UTF8, "application/json"))))
                 {
-                    var notFoundError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "NOT_FOUND" && e.path.Contains("repository"));
-                    if (notFoundError != null)
+                    if (response.StatusCode == HttpStatusCode.Unauthorized || response.StatusCode == HttpStatusCode.Forbidden)
                     {
-                        // owner/repo doesn't exist or you don't have access to the repo
-                        _unknownRepos.TryAdd((owner, name));
+                        Log.Write(await response.Content.ReadAsStringAsync());
+                        _fatalError = Errors.GitHubApiFailed(response.StatusCode.ToString());
+                        return (_fatalError, default);
                     }
 
-                    var rateLimitError = grahpApiResponse.errors.FirstOrDefault(e => e.type == "MAX_NODE_LIMIT_EXCEEDED" || e.type == "RATE_LIMITED");
-                    if (rateLimitError != null)
-                    {
-                        _rateLimitError = Errors.GitHubApiFailed(rateLimitError.message);
-                    }
+                    var content = await response.EnsureSuccessStatusCode().Content.ReadAsStringAsync();
 
-                    return (Errors.GitHubApiFailed(notFoundError?.message ?? rateLimitError?.message ?? grahpApiResponse.errors.First().message), null);
-                }
+                    var body = JsonConvert.DeserializeAnonymousType(
+                        content,
+                        new { data = default(T), errors = new[] { new { type = "", message = "" } } });
 
-                var githubUsers = new List<GitHubUser>();
-                if (grahpApiResponse.data?.repository?.@object?.history?.nodes != null)
-                {
-                    foreach (var node in grahpApiResponse.data?.repository?.@object?.history?.nodes)
+                    if (body.errors != null)
                     {
-                        if (node.author != null)
+                        foreach (var error in body.errors)
                         {
-                            githubUsers.Add(new GitHubUser
+                            switch (error.type)
                             {
-                                Id = node.author.user?.databaseId,
-                                Login = node.author.user?.login,
-                                Name = string.IsNullOrEmpty(node.author.user?.name) ? node.author.user?.login : node.author.user?.name,
-                                Emails = new[] { node.author.email },
-                            });
+                                case "NOT_FOUND":
```

Comment:
>NOT_FOUND [](start = 38, length = 9)

NOT_FOUND for repository and user need different ways to handle.

Reply:
yep